File: app/src/main/java/lv/luhmirin/sampleapp/list/ShoppingListAdapter.java
package lv.luhmirin.sampleapp.list;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import lv.luhmirin.sampleapp.R;
import lv.luhmirin.sampleapp.list.model.ShoppingItem;

class ShoppingListAdapter extends RecyclerView.Adapter<ShoppingListAdapter.ShoppingItemHolder> {

    private final Context context;

    private List<ShoppingItem> items;

    ShoppingListAdapter(@NonNull Context context) {
        this.context = context;
    }

    void setItems(List<ShoppingItem> items) {
        this.items = items;
        notifyDataSetChanged();
    }

    @Override
    public ShoppingItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ShoppingItemHolder(LayoutInflater.from(context).inflate(R.layout.item_shopping_list, parent, false));
    }

    @Override
    public void onBindViewHolder(ShoppingItemHolder holder, int position) {
        ShoppingItem item = items.get(position);

        holder.name.setText(item.getName());
        holder.price.setText(String.format("%.2f", item.getPrice()));
        holder.description.setText(item.getDescription());
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class ShoppingItemHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.item_name) public TextView name;
        @BindView(R.id.item_price) public TextView price;
        @BindView(R.id.item_description) public TextView description;

        ShoppingItemHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

}

File: app/src/main/kotlin/lv/luhmirin/sampleapp/list/ShoppingListAdapter.kt
package lv.luhmirin.sampleapp.list

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import com.pawegio.kandroid.inflateLayout
import kotlinx.android.synthetic.main.item_shopping_list.view.*
import lv.luhmirin.sampleapp.R
import lv.luhmirin.sampleapp.list.model.ShoppingItem

internal class ShoppingListAdapter(private val context: Context, var items: List<ShoppingItem> = emptyList()) : RecyclerView.Adapter<ShoppingListAdapter.ShoppingItemHolder>() {

    fun setItemsAndNotify(items: List<ShoppingItem>) {
        this.items = items
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ShoppingItemHolder {
        return ShoppingItemHolder(context.inflateLayout(R.layout.item_shopping_list))
    }

    override fun onBindViewHolder(holder: ShoppingItemHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    internal class ShoppingItemHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bind(item: ShoppingItem) {
            itemView.item_name.text = item.name
            itemView.item_price.text = "%.2f".format(item.price)
            itemView.item_description.text = item.description
        }
    }

}
